### ISSUE TITLE
Pre-trained weights + questions on input format

### ISSUE THREAD

**Issue by chiaradivece** [pakheiyeung/PlaneInVol#4]
Hey there,

Great work! I have a few questions about the implementation details:

- Input format: Could you specify the format of the volume used as input for training?
- Training with slices: According to the reply to another issue, masks are used during inference only, can individual slices be inputted during training? If possible, how should these slices be prepared? How should I modify the dataset.py file to do so?
- Inference requirements: For inference, could you elaborate on how the mask of a slice in the volume is utilized? Are there specific considerations or settings needed?
- Access to pre-trained weights: Are there pre-trained weights available for this model? 

Thank you for your help and for sharing this work!

Best,
Chiara

**Comment by pakheiyeung** [comment]
Thanks for your questions.
 
1. I am sorry that due to the data agreement, we cannot make the pre-trained weights public currently. We would like to do that in the future and are looking for alternative ways for that.
2. The volumes we tested on are 160x160x160. But other dimensions should also be fine (better to be divisible by 2^4 due to the downsampling in the network). Different file formats are fine, but maybe you need to edit the dataloader a bit.
3. The masks are used solely during training. I think in high level, as long as you can sample 2D slices from a volume and know their position in the 3D volume. You can use them for training
4. Masks are not needed during inference. They are used in testing in the paper so that we can focus on the brain region for quantifying the results. But for real applications, masks are not needed during inference.

**Comment by zikai1** [comment]
Hi, 

Thanks for sharing such a great work! I would like to request the training data and the pre-trained weights. Could you please share them with the community or by an alternative way? Thank you so much!

Best,
BIGTONE

**Comment by StormArcher** [comment]
May I ask the author and the person who raised the issues, do you have any similar public datasets that have been successfully run on the code shared by the author? I ran the training on this code using the OASIS dataset, but the result was very failed. I am currently investigating the cause. @chiaradivece @zikai1

**Comment by StormArcher** [comment]
Thank you very much to the code author for sharing such an excellent work. 

Although it didn't run successfully, the ideas in this paper have been of great help to me. 
My task requirement is to find the position of 2D data in 3D. The modalities of 2D and 3D data are different. 

I don't know if this code is suitable for this task. I hope to receive everyone's guidance @chiaradivece

**Comment by pakheiyeung** [comment]
@StormArcher Thanks for your questions. Unfortunate, I do not know any similar public dataset. My guess on why it doesn't work: (1) the data you used for training are not well co-registered to the same space (2) there is a significant domain gap between your 2D and 3D data
